### PR TITLE
[parallel execution] only bump incarnation when the task is aborted

### PR DESCRIPTION
**Pull request by zekun000**
This makes the code more consistent, no need to read incarnation number when execute.

### CODE REVIEW

File: aptos-move/parallel-executor/src/scheduler.rs [hunk 1]
```diff
@@ -81,6 +81,7 @@ pub enum SchedulerTask<'a> {
 ///    ↓                finish_abort
```
Comment (gelash) [inline aptos-move/parallel-executor/src/scheduler.rs:81]: line 74 in particular, can you update the comment please?

**Review by gelash — APPROVED** [review]
Thanks Zekun, looks a lot better like this.

File: aptos-move/parallel-executor/src/scheduler.rs (hunk 1 above)
Comment (zekun000) [reply aptos-move/parallel-executor/src/scheduler.rs:81]: good catch!

**Pull request closed without merging**